ArcadiaParties-SPA/src/app/services/token-interceptor.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpRequest, HttpInterceptor, HttpHandler, HttpEvent, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
+
+import { AuthService } from './auth.service';
+
+@Injectable()
+export class TokenInterceptor implements HttpInterceptor {
+  constructor(private authService: AuthService) {}
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return this.authService.acquireTokenResilient().pipe(
```

Comment:
And what happens if you need to login inside? I suppose, the request will never be sent.

Reply:
Fixed according to your advice